buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy

```diff
@@ -1139,7 +1140,25 @@ artifactId=${project.name}
         outputs.upToDateWhen { false }
 
         include "**/*IT.class"
-        systemProperties.beamTestPipelineOptions = configuration.integrationTestPipelineOptions
+
+        def pipelineOptionString = configuration.integrationTestPipelineOptions
+        if(configuration.runner?.equalsIgnoreCase('dataflow')) {
+          project.evaluationDependsOn(":beam-runners-google-cloud-dataflow-java-legacy-worker")
+
+          def jsonSlurper = new JsonSlurper()
+          def allOptionsList = jsonSlurper.parseText(pipelineOptionString)
+          def dataflowWorkerJar = project.findProperty('dataflowWorkerJar') ?:
+                  project.project(":beam-runners-google-cloud-dataflow-java-legacy-worker").shadowJar.archivePath
```

Comment:
Will this line only get the path to a previously built worker (without building) or it will build it here if it's not found?

Reply:
The dependency on the archive below via `shadow it.project(path: ":beam-runners-google-cloud-dataflow-java-legacy-worker", configuration: 'shadow')` will ensure it gets built.

Is there a reason why you went with a configuration based dependency over using `dependsOn ":beam-runners-google-cloud-dataflow-java-legacy-worker:shadowJar"` task like the others that were done?

Reply:
Only reason is to follow the convention in this file, for better readability. 

If there is reason other one of using `depends on` is preferred, please let me know, I shall update accordingly.